src/main/java/com/gojek/beast/protomapping/BQClient.java

```diff
@@ -65,15 +44,18 @@ public void upsertTable() throws ProtoNotFoundException, BigquerySchemaMappingEx
         log.info("Successfully upserted bigquery table");
     }
 
-    private TableDefinition getTableDefinition(Schema schema) {
+    private TableDefinition getTableDefinition(Schema schema) throws BQPartitionKeyNotSpecified {
         TableDefinition tableDefinition = StandardTableDefinition.newBuilder()
                 .setSchema(schema)
                 .build();
 
 
-        if (appConfig.isBQTablePartitioningEnabled()) {
+        if (bqConfig.isBQTablePartitioningEnabled()) {
             TimePartitioning.Builder timePartitioningBuilder = TimePartitioning.newBuilder(TimePartitioning.Type.DAY);
-            timePartitioningBuilder.setField(appConfig.getBQTablePartitionKey());
+            if (bqConfig.getBQTablePartitionKey() == null) {
+                throw new BQPartitionKeyNotSpecified(String.format("Partition key not specified for the table: %s", bqConfig.getTable()));
+            }
+            timePartitioningBuilder.setField(bqConfig.getBQTablePartitionKey());
```

Comment:
As discussed, time partitioning should be determined based on the data type of the field. For now, we could keep just timestamp based however the code should be extensible for any other data type.